db/deploy/mutations/submit_application.sql

```diff
@@ -51,6 +53,10 @@ begin
     raise 'The application cannot be submitted as the submission field submission_date is null or empty';
   end if;
 
+  if acknowledgements_array_length <> 17 or acknowledgements_array_length is null then
```

Comment:
magic number `17` :magic_wand: 

Let's add an acceptance criteria to #471 to refactor this so that we get the number of acknowledgements from the schema instead of hardcoding it here